src/utils/viewer/ObjectPickingHelper.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Renderbuffer.h>
+#include <Magnum/Magnum.h>
+#include <memory>
+#include "esp/gfx/Drawable.h"
+#include "esp/gfx/MeshVisualizerDrawable.h"
+
+class ObjectPickingHelper {
+ public:
+  /**
+   * @brief Constructor
+   * @param viewportSize the size of the viewport in w x h format
+   */
+  ObjectPickingHelper(Magnum::Vector2i viewportSize);
+  ~ObjectPickingHelper(){};
+
+  /**
+   * @brief  Recreate object ID reading renderbuffers that depend on viewport
+   * size
+   * @param viewportSize the size of the viewport in w x h format
+   */
+  ObjectPickingHelper& recreateFramebuffer(Magnum::Vector2i viewportSize);
+
+  /**
+   *@brief Prepare to draw: it will bind the framebuffer, map color attachement,
+   *clear depth, background color
+   */
+  ObjectPickingHelper& prepareToDraw();
+
+  /**
+   * @brief set viewport for framebuffer, which must be called in viewport event
+   */
+  ObjectPickingHelper& handleViewportChange(Magnum::Vector2i viewportSize);
+  /**
+   * @brief get the object id, aka, the drawable id defined in Drawable class
+   * @param eventPosition, mouse event position
+   * @param windowSize, the size of the GUI window
+   */
+  unsigned int getObjectId(Magnum::Vector2i mouseEventPosition,
+                           Magnum::Vector2i windowSize);
+
+  /**
+   * @brief create a mesh visualizer (a drawable) for the picked object
+   * @param pickedObject, the picked object (drawable)
+   */
+  ObjectPickingHelper& createPickedObjectVisualizer(
+      esp::gfx::Drawable* pickedObject);
+
+  /**
+   * @brief return true if an object is picked at the moment
+   */
+  bool objectPicked() { return meshVisualizerDrawable_ != nullptr; }
```

Comment:
```suggestion
  bool isObjectPicked() const { return meshVisualizerDrawable_ != nullptr; }
```

Not sure about the naming (`getIs`?), but if everything has `get`, this should have something also. Oh, and `const`.

Reply:
Let just use isObjectPicked(), one exception for the getter here.
Otherwise it is wrong in terms of grammar.

Yes, const will be added.